src/ldpl.cpp

```diff
@@ -1091,371 +1094,372 @@ void compile_line(vector<string> & tokens, unsigned int line_num, compiler_state
+        state.add_code(get_c_variable(state, tokens[3]) + " = localtime(&ldpl_time)->tm_mon + 1;", state.where);
         return;
     }
     if(line_like("GET MONTH IN $num-var", tokens, state))
     {
-        if(!in_procedure_section(state, line_num, current_file))
-            error("GET MONTH statement outside PROCEDURE section (\033[0m" + current_file + ":"+ to_string(line_num)+"\033[1;31m)");
+        if(!in_procedure_section(state))
+            badcode("GET MONTH statement outside PROCEDURE section", state.where);
         //C++ Code
-        state.add_code(get_c_variable(state, tokens[3]) + " = localtime(&ldpl_time)->tm_mon + 1;", line_num);
+        state.add_code(get_c_variable(state, tokens[3]) + " = localtime(&ldpl_time)->tm_mon + 1;", state.where);
         return;
     }
     if(line_like("GET EPOCH IN $num-var", tokens, state))
     {
-        if(!in_procedure_section(state, line_num, current_file))
-            error("GET EPOCH statement outside PROCEDURE section (\033[0m" + current_file + ":"+ to_string(line_num)+"\033[1;31m)");
+        if(!in_procedure_section(state))
+            badcode("GET EPOCH statement outside PROCEDURE section", state.where);
         //C++ Code
-        state.add_code(get_c_variable(state, tokens[3]) + " = ldpl_time;", line_num);
+        state.add_code(get_c_variable(state, tokens[3]) + " = ldpl_time;", state.where);
         return;
     }
     // Custom Statements
     if(line_like("CREATE STATEMENT $string EXECUTING $subprocedure", tokens, state))
     {
-        if(!in_procedure_section(state, line_num, current_file))
-            error("CREATE STATEMENT statement outside PROCEDURE section (\033[0m" + current_file + ":"+ to_string(line_num)+"\033[1;31m)");
+        if(!in_procedure_section(state))
+            badcode("CREATE STATEMENT statement outside PROCEDURE section", state.where);
         if(state.closing_subprocedure())
-            error("CREATE STATEMENT statement inside SUB-PROCEDURE (\033[0m" + current_file + ":"+ to_string(line_num)+"\033[1;31m)");
+            badcode("CREATE STATEMENT statement inside SUB-PROCEDURE", state.where);
         else if(state.closing_if())
-            error("CREATE STATEMENT statement inside IF (\033[0m" + current_file + ":"+ to_string(line_num)+"\033[1;31m)");
+            badcode("CREATE STATEMENT statement inside IF", state.where);
         else if(state.closing_loop())
-            error("CREATE STATEMENT statement inside WHILE or FOR (\033[0m" + current_file + ":"+ to_string(line_num)+"\033[1;31m)");
+            badcode("CREATE STATEMENT statement inside WHILE or FOR", state.where);
         string model_line = tokens[2].substr(1, tokens[2].size() - 2);
         vector<string> model_tokens;
         vector<string> parameters = state.subprocedures[tokens[4]];
         trim(model_line);
-        tokenize(model_line, 0, model_tokens, state.current_file, true, ' ');
+        state.where.current_line = 0; // XXX why ?
```

Comment:
You can delete this line, 0 as the line number was wrong.

Reply:
done